### DISCUSSION TITLE
Using vsce to publish a VSIX with a different version than the one it was packaged with?

### DISCUSSION THREAD

**Discussion by jakereifer** [microsoft/vscode-discussions#828] [Extension Development QnA] [answered]
From my experience, it seems that if you package an extension into a VSIX using _vsce_ and then publish that VSIX using _vsce_ (but supply a different version), it doesn't work as expected.

For instance: 
`vsce package 1.1.1` creates `helloworld-1.1.1.vsix`

`vsce publish --packagePath helloworld-1.1.1.vsix 1.1.2` doesnt work:

```
DONE  Published prospect-test-ex.helloworldprospecttest v1.1.1.
ERROR  ENOENT: no such file or directory, open '1.1.2'
```

Is there a way to publish a VSIX using _vsce_ but supply a different version than the one it was packaged with?

**Comment by Ninglo** [comment, marked as answer]
I'm afraid this won't work. `vsce package <version>` not only packages the plug-in, but also adds `extension.vsixmanifest` and other information to the product. In other words, version-related information has been defined in `helloworld-1.1.1.vsix`, so it is unreasonable to release this product as version `1.1.2`.
(If you want to know the details of packaging, you can change the `vsix` extension to `zip` and unzip it to view)

**Reply by burkeholland** [reply]
Marking this as answer since we're past two weeks here. If you feel this is not the case, please feel free to re-open.
